src/debug.c

```diff
@@ -1820,83 +1820,79 @@ void closeDirectLogFiledes(int fd) {
     if (!log_to_stdout) close(fd);
 }
 
+#if defined(HAVE_BACKTRACE) && defined(__linux__)
+static int stacktrace_pipe[2] = {0};
+void setupStacktracePipe(void) {
+    if (-1 == pipe2(stacktrace_pipe, O_CLOEXEC | O_NONBLOCK)) {
```

Comment:
Can we reuse `anetPipe()` here same as other pipes? Seems like it has some portability fixes in it: https://github.com/redis/redis/blob/3d9c427f8c017c62e3e62883c4dfe599f060f104/src/anet.c#L634

Reply:
We don't need the pipe on other systems than linux.

Reply:
regardless of portability, we have a convenience wrapper, let's use it.

Reply:
but why initialize the pipe for systems where it will never be used?

Reply:
I meant portability between linux versions. It tries pipe2() first then fallback to pipe(). Also, in future, if we want to introduce a fix to pipe2(), then we can do it in a single place if all places call single wrapper function.

Reply:
> but why initialize the pipe for systems where it will never be used?

it doesn't do that. it's just that when it does create a pipe, it does it with our wrapper, same one we use everywhere else..